rust/src/action.rs

```diff
@@ -150,7 +173,7 @@ pub struct Add {
     /// The size of this file in bytes
     pub size: DeltaDataTypeLong,
     /// A map from partition column to value for this file
-    pub partition_values: HashMap<String, String>,
+    pub partition_values: HashMap<String, Option<String>>,
```

Comment:
It occurred to me that the problem you found has a much bigger impact than just the partition_values field, it also applies to Add.tags, Remove.tags, MetaData.format.options, MetaData.configurations.

Let me know if you prefer to address them or only focus on partition_values in this PR. If you prefer the latter, I can file an issue to track the fix for the other fields.

Reply:
Hey QP, I went through the spec doc but didn't found situations that those structs could also contain null values in the map. Is that because Map[String, String] allows null value in Scala, so that we also need to make the change to HashMap<String, Option<String>>?

Reply:
That's correct, based on the discussion in the dev list, all data types are annotated using scala syntax, so all `Map[String, String]` should map to `HashMap<String, Option<String>>` in rust.

Reply:
btw, I am perfectly fine with us addressing these other fields in an other PR if you want to keep the scope of this PR small. The fix for partition_values itself is already a  big improvement :)

Reply:
Let me take a look on this :)